lighthouse-logger/test/index-test.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+/* eslint-disable no-invalid-this */
+
+const assert = require('assert');
+const log = require('..');
+
+afterEach(log.takeTimeEntries);
+
+describe('Log.timeDecorate', function() {
+  it('records timings', () => {
+    const fn = () => {};
+    const decoratedFn = log.timeDecorate(fn, {
+      msg: 'msg',
+      id: 'id',
+    });
+    decoratedFn();
+    assert.equal(log.getTimeEntries().length, 1);
+    assert.ok(log.getTimeEntries()[0].duration > 0);
+  });
+
+  it('retains return value', () => {
+    const fn = () => 'works';
+    const decoratedFn = log.timeDecorate(fn, {
+      msg: 'msg',
+      id: 'id',
+    });
+    assert.equal(decoratedFn(), 'works');
+  });
+
+  it('retains async return value', async () => {
+    const fn = async () => 'works';
+    const decoratedFn = log.timeDecorate(fn, {
+      msg: 'msg',
+      id: 'id',
+    });
+    assert.equal(await decoratedFn(), 'works');
+  });
+
+  it('retains parameters', () => {
+    const fn = (value1, value2) => value1 + value2;
+    const decoratedFn = log.timeDecorate(fn, {
+      msg: 'msg',
+      id: 'id',
+    });
+    assert.equal(decoratedFn(1, 2), 3);
+  });
+
+  it('retains this binding', () => {
+    const object = new function() {
+      this.value = 'works';
+      this.getValue = () => {
+        return this.value;
+      };
+    };
+    const fn = object.getValue;
+    const decoratedFn = log.timeDecorate(fn, {
+      msg: 'msg',
+      id: 'id',
+    });
+    assert.equal(decoratedFn(), 'works');
+  });
+
+  it('accepts function template for msg and id', () => {
+    // eslint-disable-next-line no-unused-vars
```

Comment:
maybe assert the values are what you expect them to be or just get rid of them if they're not important to the test?